Firestore/core/test/unit/bundle/bundle_reader_test.cc

```diff
@@ -0,0 +1,533 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/bundle/bundle_reader.h"
+
+#include "Firestore/Protos/cpp/firestore/bundle.pb.h"
+#include "Firestore/Protos/cpp/firestore/local/maybe_document.pb.h"
+#include "Firestore/Protos/cpp/google/firestore/v1/document.pb.h"
+#include "Firestore/core/src/bundle/named_query.h"
+#include "Firestore/core/src/core/field_filter.h"
+#include "Firestore/core/src/local/local_serializer.h"
+#include "Firestore/core/src/model/database_id.h"
+#include "Firestore/core/src/nanopb/byte_string.h"
+#include "Firestore/core/src/nanopb/message.h"
+#include "Firestore/core/src/remote/serializer.h"
+#include "Firestore/core/test/unit/nanopb/nanopb_testing.h"
+#include "Firestore/core/test/unit/testutil/status_testing.h"
+#include "Firestore/core/test/unit/testutil/testutil.h"
+#include "google/protobuf/util/json_util.h"
+#include "google/protobuf/util/message_differencer.h"
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+namespace bundle {
+namespace {
+
+using google::protobuf::util::MessageDifferencer;
+using google::protobuf::util::MessageToJsonString;
+using ProtoBundledDocumentMetadata = ::firestore::BundledDocumentMetadata;
+using ProtoBundleElement = ::firestore::BundleElement;
+using ProtoBundleMetadata = ::firestore::BundleMetadata;
+using ProtoDocument = ::google::firestore::v1::Document;
+using ProtoMaybeDocument = ::firestore::client::MaybeDocument;
+using ProtoNamedQuery = ::firestore::NamedQuery;
+using ProtoValue = ::google::firestore::v1::Value;
+using model::DatabaseId;
+using nanopb::ProtobufParse;
+
+class BundleReaderTest : public ::testing::Test {
+ public:
+  BundleReaderTest()
+      : remote_serializer(DatabaseId("p", "default")),
+        local_serializer(remote_serializer),
+        bundle_serializer(remote_serializer) {
+    msg_diff_.ReportDifferencesToString(&message_differences);
+  }
+
+  static std::string FullPath(const std::string& path) {
+    return "projects/p/databases/default/documents/" + path;
+  }
+
+  std::string AddNamedQuery(const ProtoNamedQuery& data) {
```

Comment:
Nit: include the standard headers used in this file (`<string>`, `<utility>`, `<memory>`, `<vector>`, `<sstream>`).